generators/entity/generator.mjs

```diff
@@ -149,7 +150,10 @@ export default class EntityGenerator extends BaseGenerator {
         this.loadClientConfig(undefined, this.application);
         this.loadTranslationConfig(undefined, this.application);
         // Try to load server config from microservice side, falling back to the app config.
-        this.loadServerConfig(getConfigWithDefaults({ ...this.jhipsterConfig, ...this.microserviceConfig }), this.application);
+        this.loadServerConfig(
+          getConfigWithDefaults({ ...deepCleanup(this.jhipsterConfig), ...deepCleanup(this.microserviceConfig ?? {}) }),
```

Comment:
{...deepCleanup(getConfigWithDefaults(this.jhipsterConfig, this.microserviceConfig ?? {}))}

Reply:
Also, should be added to all load, shouldn't it?

Reply:
> {...deepCleanup(getConfigWithDefaults(this.jhipsterConfig, this.microserviceConfig ?? {}))}

No, cleanup should be applied before spreading and before defaults.

> Also, should be added to all load, shouldn't it?

Yes, but in this case my plan is to create a custom storage that applies defaults at read. So jhipsterConfig will have defaults applied and jhipsterConfigWithDefaults will be dropped.
This will be another PR.

Reply:
forget about it